snooty/parser.py

```diff
@@ -1046,11 +1047,23 @@ def _locate_text(text: str) -> int:
             image_argument = options.get("icon")
             url_argument = options.get("url")
 
-            if image_argument:
-                self.validate_and_add_asset(doc, image_argument, line)
             if url_argument and not url_argument.startswith("http"):
                 self.validate_relative_url(url_argument, line)
 
+            # for cards - if the image is a path, we need a dark mode version as well
+            if key == "mongodb:card":
+                if image_argument and "/" in image_argument:
+                    self.diagnostics.append(CardIconString(image_argument, line))
+                    self.validate_and_add_asset(doc, image_argument, line)
+                    dark_mode_image_argument = options.get("icon-dark")
+                    if dark_mode_image_argument:
+                        self.validate_and_add_asset(doc, dark_mode_image_argument, line)
+                    else:
+                        self.diagnostics.append(ExpectedOption(name, "icon-dark", line))
+            else:
+                if image_argument:
+                    self.validate_and_add_asset(doc, image_argument, line)
+
```

Comment:
total nit: you could consolidate a tiny bit more if you wanted, I believe:
```
if key == "mongodb:card" and image_argument:
  self.validate_and_add_asset(doc, image_argument, line)
  if "/" in image_argument:
    ... the rest of that block of code (but with the validate_and_add_asset line unneeded now
```

Reply:
i do want to consolidate but i think the `validate_and_add_asset()` function should only be called on path arguments (i.e. with arguments with "/") :(

Reply:
Correct -- `validate_and_add_asset()` flags a path to be read down the road -- it gets turned into a concrete Path down the call chain (this is a little bit of a code smell).

Reply:
aha, I didn't see the difference in the arguments! Thanks!